snuba/clickhouse/formatter/expression.py

```diff
@@ -154,4 +154,34 @@ def visit_argument(self, exp: Argument) -> str:
     def visit_lambda(self, exp: Lambda) -> str:
         parameters = [self.__escape_identifier_enforce(v) for v in exp.parameters]
         ret = f"({', '.join(parameters)} -> {exp.transformation.accept(self)})"
-        return self.__alias(ret, exp.alias)
+        return self._alias(ret, exp.alias)
+
+
+class ClickHouseExpressionFormatterAnonymized(ClickhouseExpressionFormatter):
+    """
+    This Formatter strips string and integer literals and replaces them with a
+    a token representing the type of literal. All other behavior is the same.
+    """
+
+    def visit_literal(self, exp: Literal) -> str:
+        if exp.value is None:
+            return self._alias("NULL", exp.alias)
+        elif exp.value is True:
+            return self._alias("true", exp.alias)
+        elif exp.value is False:
+            return self._alias("false", exp.alias)
+        elif isinstance(exp.value, str):
```

Comment:
i think i can remove the duplication in this function here, going to try that real quick.